### ISSUE TITLE
Blazorise AODA compatibility

### ISSUE THREAD

**Issue by SukhdeepCao** [Megabit/Blazorise#3454]
As I am using Blazorise component library for building Blazor server application.
but I don't see blazorise add the required attributes for AODA compatability.

**Comment by stsrki** [comment]
Can you please explain what is AODA compatibility?

**Comment by SukhdeepCao** [comment]
AODA stands for Accessibility for Ontarians with Disabilities Act.
Under this act, web page must be accessible to all disable people so that they can listen to the page with the help of page reader and perform actions accordingly.

when Blazorise component renders it should add the ARIA attributes for example **aria-label**, **aria-describedby** based on the component.

for further reference:
https://www.aoda.ca/
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA

**Comment by stsrki** [comment]
Do you mean it should render something by default in `aria-label`, `aria-describedby` attributes? Are there any guidelines what should be the content of those attributes?

**Comment by SukhdeepCao** [comment]
`aria-label`, `aria-describedby` just an example, but yes it should render aria attributes based on the component.
if we take an example of **Bootstrap**, if we use directly HTML code from bootstrap, they have added all the aria attributes there in the HTML code.
plus, if we use the default Blazor component like InputText, it adds the` aria-invalid `true or false based on the value. 

Therefore, Blazorise should compliance with accessibility by default.

**Comment by stsrki** [comment]
https://www.a11yproject.com/

**Comment by stsrki** [comment]
Tool for checking the compliance https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh

**Comment by stsrki** [comment]
https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-6.0#focus-an-element-on-navigation

**Comment by stsrki** [comment]
https://denodell.com/blog/html-best-kept-secret-output-tag

**Comment by stsrki** [comment]
https://accented.dev/
